cpp/include/ray/api/ray_config.h

```diff
@@ -9,45 +9,31 @@
 namespace ray {
 namespace api {
 
-enum class RunMode { SINGLE_PROCESS, CLUSTER };
-
-/// TODO(Guyang Song): Make configuration complete and use to initialize.
 class RayConfig {
  public:
-  WorkerType worker_type = WorkerType::DRIVER;
-
-  RunMode run_mode = RunMode::SINGLE_PROCESS;
-
-  std::string redis_ip;
-
-  int redis_port = 6379;
-
-  std::string redis_password = "5241590000000000";
+  // The redis address of existing cluster.
+  std::string address;
 
-  int node_manager_port = 62665;
+  // Whether or not to run this program in a single process. This is used for debugging.
+  bool local_mode = true;
 
-  std::string lib_name = "";
+  // The dynamic library path which contains remote fuctions of users.
+  // Local mode does not need to set this parameter.
+  std::string dynamic_library_path = "";
 
-  std::string store_socket = "";
+  /* The following are unstable parameters and their use is discouraged. */
 
-  std::string raylet_socket = "";
+  // Prevent external clients without the password from connecting.
+  std::string redis_password_ = "5241590000000000";
 
-  std::string session_dir = "";
-
-  static std::shared_ptr<RayConfig> GetInstance();
-
-  void SetRedisAddress(const std::string address) {
-    auto pos = address.find(':');
-    if (pos == std::string::npos) {
-      throw RayException("The address has no char ':'");
-    }
-
-    redis_ip = address.substr(0, pos);
-    redis_port = std::stoi(address.substr(pos + 1, address.length()));
+  // Set argc and argv from command line. The params in argv will be parsed by `gflag`.
+  // And Command line parameters have higher priority.
```

Comment:
```suggestion
  // Update config values based on command-line arguments. If a parameter is explicitly set in command-line arguments, the parameter value will be updated, otherwise, the parameter value will stay unchanged.
```